query-template/src/template.rs

```diff
@@ -0,0 +1,54 @@
+use crate::fragment::Fragment;
+use crate::placeholder::PlaceholderFormat;
+use std::fmt;
+use std::fmt::Debug;
+
+#[derive(Debug)]
+pub struct QueryTemplate<P> {
+    pub fragments: Vec<Fragment>,
+    pub parameters: Vec<P>,
+    pub placeholder_format: PlaceholderFormat,
+}
+
+impl<P> QueryTemplate<P> {
+    pub fn new(placeholder: PlaceholderFormat) -> Self {
+        QueryTemplate {
+            fragments: Vec::new(),
+            parameters: Vec::new(),
+            placeholder_format: placeholder,
+        }
+    }
+}
+
+const BEGIN_REPEAT: &str = "/* prisma-comma-repeatable-start */";
+const END_REPEAT: &str = "/* prisma-comma-repeatable-end */";
```

Comment:
Why do we still need these?

Reply:
These were needed only to confirm to existing test snaps. Searching for them now again, but found none.

We can just use the `[?]` notation instead in test snaps.

Reply:
The test snaps don't use this code path, they are formatted via a different implementation in `query-engine/query-builders/query-builder/src/lib.rs`.

Reply:
That's right. It appears that I confused the two. My bad.

Reply:
Resolved this. The `ParameterizedRow` has to store the number of items, it becomes part of the `QueryTemplate`.

(Some tests still do not pass.)